src/one/nio/ws/exception/HandshakeException.java

```diff
@@ -0,0 +1,11 @@
+package one.nio.ws.exception;
+
+/**
+ * @author <a href="mailto:vadim.yelisseyev@gmail.com">Vadim Yelisseyev</a>
+ */
+public class HandshakeException extends RuntimeException {
```

Comment:
Is it possible to extend `WebSocketException` here as siblings do?

Reply:
I think the handshake happens before the request goes from HTTP to WebSocket state so I separated them in two different hierarchies. But I don't mind to join it :)